src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -0,0 +1,120 @@
+#include "software/backend/simulation/physics/physics_robot.h"
+
+#include "shared/constants.h"
+#include "software/backend/simulation/physics/box2d_util.h"
+
+PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot)
+    : robot_id(robot.id())
+{
+    // All the BodyDef must be defined before the body is created.
+    // Changes made after aren't reflected
+    createRobotPhysicsBody(world, robot);
+    setupRobotPhysicsBody(robot);
+    setupChickerPhysicsBody(robot);
+}
+
+PhysicsRobot::~PhysicsRobot()
+{
+    // Examples for removing bodies safely from
+    // https://www.iforce2d.net/b2dtut/removing-bodies
+    b2World* robot_body_world = robot_body->GetWorld();
+    if (bodyExistsInWorld(robot_body, robot_body_world))
+    {
+        robot_body_world->DestroyBody(robot_body);
+    }
+}
+
+void PhysicsRobot::createRobotPhysicsBody(std::shared_ptr<b2World> world,
+                                          const Robot& robot)
+{
+    robot_body_def.type = b2_dynamicBody;
+    robot_body_def.position.Set(robot.position().x(), robot.position().y());
+    robot_body_def.linearVelocity.Set(robot.position().x(), robot.position().y());
```

Comment:
You are using position here instead of velocity.
Typo?

Reply:
massive typo. Fixed